File: Src/java/elm-jaxb/src/main/java/org/cqframework/cql/elm/serializing/jaxb/ElmLibraryReaderProvider.java
package org.cqframework.cql.elm.serializing.jaxb;

import org.cqframework.cql.elm.serializing.ElmLibraryReader;

public class ElmLibraryReaderProvider implements org.cqframework.cql.elm.serializing.ElmLibraryReaderProvider {
    @Override
    public ElmLibraryReader create(String contentType) {
        if (contentType == null) {
            contentType = "application/elm+json";
        }

        switch (contentType) {
            case "application/elm+xml":
                return new ElmXmlLibraryReader();
            case "application/elm+json":
            default:
                return new ElmJsonLibraryReader();
        }
    }
}

File: Src/java/elm-xmlutil/src/main/java/org/cqframework/cql/elm/serializing/xmlutil/ElmLibraryReaderProvider.kt
package org.cqframework.cql.elm.serializing.xmlutil

import org.cqframework.cql.elm.serializing.ElmLibraryReader
import org.cqframework.cql.elm.serializing.ElmLibraryReaderProvider

class ElmLibraryReaderProvider : ElmLibraryReaderProvider {
    override fun create(contentType: String): ElmLibraryReader {
        var contentType: String? = contentType
        if (contentType == null) {
            contentType = "application/elm+json"
        }
        return when (contentType) {
            "application/elm+xml" -> ElmXmlLibraryReader()
            "application/elm+json" -> ElmJsonLibraryReader()
            else -> ElmXmlLibraryReader()
        }
    }
}
